src/domain/Document/Document.repository.interface.ts

```diff
@@ -1,13 +1,12 @@
-import { IDocumentPersistenceDTO } from '@persistence/dtos/Document';
-import { Document } from './Document';
+import { Document, IDocumentProps } from './Document';
 
 export interface IDocumentRepository {
   getAll(
     sourcesOfInterest?: string[],
     offset?: number,
     limit?: number,
   ): Promise<{ entries: Document[]; count: number }>;
-  save(dto: IDocumentPersistenceDTO): Promise<Document>;
-  update(dto: IDocumentPersistenceDTO): Promise<Document>;
+  save(dto: IDocumentProps): Promise<Document>;
```

Comment:
I would move IDocumentProps in the same file with the abstraction.

Reply:
Since the props are used in the constructors of the entities, I believe we should let them there for the moment